File: DSA/SuryanshMishra_590012069/Day33/day33question1.cpp
// 1. Check if Two Numbers Have Opposite Signs
// A fundamental bitwise check that determines if two integers have opposite signs by using the sign bit. This is commonly used in systems programming, signal handling, and even game logic.

// Given two integers a and b, use bitwise operations to check if they have opposite signs. This works because the sign bit (most significant bit) will differ if the signs are different.

// This teaches sign bit manipulation, XOR/sign checking, and is useful in low-level debugging or optimization.

// Your task: Check whether two numbers have opposite signs using bitwise operations.

// Examples
// Input:

// a = 2, b = -3
// Output:

// True
// Input:

// a = -4, b = -5
// Output:

// False
#include <iostream>
using namespace std;
int main() {
int a, b;
cin >> a >> b;
bool oppositeSigns = (a ^ b) < 0;
cout << (oppositeSigns ? "True" : "False") << endl;
return 0;
}

File: DSA/Pranvkumar_590011587/Day33_Question1.py
def have_opposite_signs(a, b):
    """
    Check if two numbers have opposite signs.
    Opposite signs mean a XOR b is negative.
    """
    return (a ^ b) < 0

print("1. Check if Two Numbers Have Opposite Signs")
print("Input: a = 2, b = -3 -> Output:", have_opposite_signs(2, -3)) 
print("Input: a = -4, b = -5 -> Output:", have_opposite_signs(-4, -5)) 